workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/views/matplotlib.py

```diff
@@ -89,7 +102,71 @@ def plot_2d_matplotlib(
         template.render(
             image_data=data,
             runs=sorted(run_diags.runs),
-            variables_to_plot=variables_to_plot,
+            variables_to_plot=sorted(variables_to_plot),
+            varfilter=varfilter,
+        )
+    )
+
+
+def plot_cube_matplotlib(
+    run_diags: RunDiagnostics,
+    run_metrics: pd.DataFrame,
+    varfilter: str,
+    metric_names: Mapping[str, str] = None,
+    **opts,
+) -> str:
+    """Plot horizontal maps of cubed-sphere data for all diagnostics whose name includes
+    varfilter. All matching diagnostics must have tile, x and y dimensions
+    and each dataset in run_diags must include lat/lon/latb/lonb coordinates."""
+
+    data = defaultdict(dict)
+
+    variables_to_plot = [
+        varname for varname in run_diags.variables if varfilter in varname
+    ]
+
+    for run in run_diags.runs:
+        grid_ds = xr.merge(
+            [run_diags.get_variable(run, varname) for varname in _COORD_VARS]
+        )
+        for varname in variables_to_plot:
+            logging.info(f"plotting {varname} in {run}")
+            v = run_diags.get_variable(run, varname)
+            ds = xr.merge([grid_ds, v])
+            plot_title = _get_map_title(
+                run_metrics[run_metrics.run == run], varname, metric_names
+            )
+            fig, ax = plt.subplots(
+                figsize=(6, 3), subplot_kw={"projection": ccrs.Robinson()}
+            )
+            mv = fv3viz.mappable_var(ds, varname, coord_vars=_COORD_VARS, **COORD_NAMES)
+            fv3viz.plot_cube(mv, ax=ax, **opts)
+            ax.set_title(plot_title)
+            plt.subplots_adjust(left=0.01, right=0.75, bottom=0.02)
+            data[varname][run] = fig_to_b64(fig)
+            plt.close(fig)
+    return raw_html(
+        template.render(
+            image_data=data,
+            runs=sorted(run_diags.runs),
+            variables_to_plot=sorted(variables_to_plot),
             varfilter=varfilter,
         )
     )
+
+
+def _get_map_title(
+    metrics: pd.DataFrame, varname: str, metric_names: Mapping[str, str],
+) -> str:
+    metric_names = {} if metric_names is None else metric_names
+    shortname = varname.split("_")[0]
+    title_parts = []
+    for name_in_figure_title, metric_prefix in metric_names.items():
+        metric_name = f"{metric_prefix}/{shortname}"
```

Comment:
This routine does too much. Most of it is related to querying the DataFrame object for certain metrics. Then it renders the result of this query into a string. You should refactor the querying and rendering to different functions. I could envision several related functions which could be grouped into a class e.g.:

```
get_metric_value(metric_type: str, variable: str, run: str) -> float
get_metric_units(metric_type: str, variable: str, run: str) -> str
get_metric_types() -> List[str]
get_metric_variables(metric_type) -> List[str]
```

I would implement some of these in `computed_diagnostics.py` and test them.

Reply:
Feels a bit weird to define a new class that is basically just a thin wrapper around a DataFrame but I agree it will be convenient.

Reply:
Agreed.  Sometimes general data containers like datasets or dataframes aren't the right abstraction. Naming/attr conventions are needed to store the data container on disk, but once we load it into to memory, it makes sense to provide the interfaces more tightly scoped to our specific application.

Reply:
I added a `RunMetrics` class.